src/types/local_date.rs

```diff
@@ -0,0 +1,268 @@
+            }
+        } else {
+            for m in (month + 1)..=12 {
+                total_days -= Self::days_in_month(m, year) as i32;
+            }
+            total_days -= Self::days_in_month(month, year) as i32 - day as i32 + 1;
+
+            for y in (year + 1)..1970 {
+                total_days -= if Self::is_leap_year(y) { 366 } else { 365 };
+            }
+        }
+
+        total_days
+    }
+
+    /// Adds the number of days to self.
+    fn add_day(&mut self, mut n: i64) {
+        while n != 0 {
+            let days_in_year = match Self::is_leap_year(self.year - (n < 0) as i32) {
+                true => 366,
+                false => 365,
+            };
+
+            if n >= days_in_year {
+                n -= days_in_year;
+                self.year += 1;
+            } else if n <= -days_in_year {
+                n += days_in_year;
+                self.year -= 1;
+            } else {
+                break;
+            }
+        }
+
+        while n != 0 {
+            let days_in_current_month = Self::days_in_month(self.month, self.year) as i64;
+
+            if n >= days_in_current_month {
+                n -= days_in_current_month;
+                self.month += 1;
+                if self.month > 12 {
+                    self.month = 1;
+                    self.year += 1;
+                }
+            } else if n <= -days_in_current_month {
+                n += days_in_current_month;
+                self.month -= 1;
+                if self.month < 1 {
+                    self.month = 12;
+                    self.year -= 1;
+                }
+            } else {
+                break;
+            }
+        }
+
+        self.day += n as i8;
+        while self.day < 1 {
+            self.month -= 1;
+            if self.month < 1 {
+                self.month = 12;
+                self.year -= 1;
+            }
+            self.day += Self::days_in_month(self.month, self.year);
+        }
+
+        while self.day > Self::days_in_month(self.month, self.year) {
+            self.day -= Self::days_in_month(self.month, self.year);
+            self.month += 1;
+            if self.month > 12 {
+                self.month = 1;
+                self.year += 1;
+            }
+        }
+    }
+
+    /// Checks whether the year is leap year.
+    fn is_leap_year(year: i32) -> bool {
+        (year % 4 == 0 && year % 100 != 0) || (year % 400 == 0)
+    }
+
+    #[napi(js_name = "toString")]
+    pub fn to_format(&self) -> String {
+        self.to_string()
+    }
+    
+    /// Returns the number of days in a month.
+    fn days_in_month(month: i8, year: i32) -> i8 {
+        match month {
+            1 | 3 | 5 | 7 | 8 | 10 | 12 => 31,
+            4 | 6 | 9 | 11 => 30,
+            2 => {
+                if Self::is_leap_year(year) {
+                    29
+                } else {
+                    28
+                }
+            }
+            _ => 0,
```

Comment:
Now this function is only used in the condition:
`if !(day >= 1 && day <= Self::days_in_month(month, year)) {`
that is after the conditional 
`if !(1..=12).contains(&month)`
So it will never be called with a non-existent month.